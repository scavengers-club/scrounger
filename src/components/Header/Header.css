header {
  background-color: var(--seafoam);
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
  height: 12vh;
  border-bottom-right-radius: 15px;
  border-bottom-left-radius: 15px;
  filter: drop-shadow(5px 5px 1px rgba(62, 62, 62, 0.5));
}

.nav-link {
  color: var(--purple);
  font-weight: 900;
  font-size: 5vw;
  text-shadow: 2px 2px 2px white;
  text-decoration: none;
}

.nav-icon {
  filter: drop-shadow(2px 2px 2px rgb(62, 62, 62));
}

.active {
  border-bottom: 5px solid var(--purple);
}
