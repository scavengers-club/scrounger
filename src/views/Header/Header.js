import React from 'react';
import { NavLink } from 'react-router-dom';
import { logOut } from '../../services/users';
import home from '../../navIcons/home.png';
import profile from '../../navIcons/profile.png';
import logout from '../../navIcons/logout.png';

export default function Header({ currentUser, setCurrentUser }) {
  const logOutUser = async () => {
    await logOut();
    setCurrentUser(null);
  };

  return (
    <header>
      <NavLink exact to="/">
        <img src={home}></img>
      </NavLink>
      {currentUser && (
        <>
          <NavLink to="/user/:id">
            <img src={profile}></img>
          </NavLink>
          <img src={logout} onClick={logOutUser}></img>
        </>
      )}
      {!currentUser && (
        <>
          <NavLink to="/register">Register</NavLink>
          <NavLink to="/login">Log In</NavLink>
        </>
      )}
    </header>
  );
}
