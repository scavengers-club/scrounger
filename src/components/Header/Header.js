import React from 'react';
import { NavLink } from 'react-router-dom';
import { logOut } from '../../services/users';
import home from '../../navIcons/home.png';
import profile from '../../navIcons/profile.png';
import logout from '../../navIcons/logout.png';
import './Header.css';

export default function Header({ currentUser, setCurrentUser }) {
  const logOutUser = async () => {
    if (confirm('are you sure you want to logout?') === true) {
      await logOut();
      setCurrentUser(null);
    }
  };

  return (
    <header>
      <NavLink exact to="/">
        <img className="nav-icon" src={home}></img>
      </NavLink>
      {currentUser && (
        <>
          <NavLink to="/profile">
            <img src={profile}></img>
          </NavLink>
          <img src={logout} onClick={logOutUser}></img>
        </>
      )}
      {!currentUser && (
        <>
          <NavLink className="nav-link" to="/register">
            Register
          </NavLink>
          <NavLink className="nav-link" to="/login">
            Log In
          </NavLink>
        </>
      )}
    </header>
  );
}
