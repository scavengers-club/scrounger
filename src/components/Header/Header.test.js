import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import Header from './Header';

test('<Header/> should display register and log in options', () => {
  const { container } = render(
    <MemoryRouter>
      <Route>
        <Header />
      </Route>
    </MemoryRouter>
  );

  screen.getByText('Register');
  expect(container).toMatchSnapshot();
});
