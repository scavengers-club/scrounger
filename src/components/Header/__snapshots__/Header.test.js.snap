// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should display register and log in options 1`] = `
<div>
  <header>
    <a
      aria-current="page"
      class="nav-link active"
      href="/"
    >
      <img
        alt="Home page"
        class="nav-icon"
        src="/nav_icons/home-icon.png"
      />
    </a>
    <a
      class="nav-link"
      href="/register"
    >
      Register
    </a>
    <a
      class="nav-link"
      href="/login"
    >
      Log In
    </a>
  </header>
</div>
`;
